adafruit_max7219/max7219.py

```diff
@@ -0,0 +1,123 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_max7219.MAX7219` - MAX7219 LED Matrix/Digit Display Driver
+============================================================
+CircuitPython library to support MAX7219 LED Matrix/Digit Display Driver.
+This library supports the use of the MAX7219-based display in CircuitPython, 
+either an 8x8 matrix or a 8 digit 7-segment numeric display.
+
+see also
+========
+* matrices.Maxtrix8x8 is a class support an 8x8 led matrix display
+* bcddigits.BCDDigits is a class that support the 8 digit 7-segment display
+
+Beware that most CircuitPython compatible hardware are 3.3v logic level! Make
+sure that the input pin is 5v tolerant.
+* Author(s): Michael McWethy
+
+Implementation Notes
+--------------------
+**Hardware:**
+* Adafruit `MAX7219CNG LED Matrix/Digit Display Driver - MAX7219 <https://www.adafruit.com/product/453>`_ (Product ID: 453)
+**Software and Dependencies:**
+* Adafruit CircuitPython firmware (1.0.0+) for the ESP8622 and M0-based boards: https://github.com/adafruit/circuitpython/releases
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+**Notes:**
+#.  Datasheet: https://cdn-shop.adafruit.com/datasheets/MAX7219.pdf
+"""
+# MicroPython SSD1306 OLED driver, I2C and SPI interfaces
+import framebuf
+import digitalio
+
+from adafruit_bus_device import spi_device
+
+# register definitions
+_DIGIT0 = const(1)
+_INTENSITY = const(10)
+
+
+class MAX7219:
+    
+    def __init__(self, w, h, spi, csPin,
+                 baudrate=8000000, polarity=0, phase=0):
+        """
+        :param int w: the number of pixels wide
+        :param int h: the number of pixels high
+        :param object spi: an spi busio or spi bitbangio object
+        :param ~microcontroller.Pin csPin: board pin to use as chip select signal
+        """
+        
+        cs = digitalio.DigitalInOut(csPin)
```

Comment:
Lets change the cspin type from microcontroller.Pin to digitalio.DigitalInOut so that other pins that implement the DigitalInOut API work too.